src/main/java/com/board/config/boardconfiggui/IpConfigController.java

```diff
@@ -66,24 +67,33 @@ public class IpConfigController implements Initializable, BoardPageDataSaverInte
 
   private BoardResultsRepo boardResultsRepo;
 
+  private PinConfig pinConfig;
+
+  private IpPinConfig sclIpConfigModel,
+          sdaIpConfigModel;
+
   private final List<IpPinConfig> ipPinConfigs = new ArrayList<>();
 
   private final List<SlaveWidgetController> slaveWidgetControllers = new ArrayList<>();
 
   @FXML
   private void onSCLPinUpdate() {
-    String sclPin = sclChoiceBox.getValue();
-    Pair<String, String> sclPinPair = getPinPortPair(sclPin);
-    ipConfigModel.setSclPort(sclPinPair.getKey());
-    ipConfigModel.setSclPin(sclPinPair.getValue());
+    IpPinConfig ipPinConfig = sclChoiceBox.getValue();
+    if (Objects.isNull(ipPinConfig)) {
+      return;
+    }
+    ipConfigModel.setSclPort(ipPinConfig.getPortName());
+    ipConfigModel.setSclPin(ipPinConfig.getPinName());
   }
 
   @FXML
   private void onSDAPinUpdate() {
-    String sdaPin = sdaChoiceBox.getValue();
-    Pair<String, String> sdaPinPair = getPinPortPair(sdaPin);
-    ipConfigModel.setSdaPort(sdaPinPair.getKey());
-    ipConfigModel.setSdaPin(sdaPinPair.getValue());
+    IpPinConfig ipPinConfig = sdaChoiceBox.getValue();
+    if (Objects.isNull(ipPinConfig)) {
```

Comment:
Don't we need to reset when the user chooses the option as "select" from the list?

Reply:
We are not allowing user to select the default option as it is mandatory field.